pkg/protocol/engine/therealledger/utxoledger/ledger.go

```diff
@@ -40,6 +40,9 @@ type Ledger struct {
 func NewProvider() module.Provider[*engine.Engine, therealledger.Ledger] {
 	return module.Provide(func(e *engine.Engine) therealledger.Ledger {
 		l := New(e.Workers.CreateGroup("Ledger"), e.Storage.Ledger(), e.API, e.Events.Error.Trigger)
+		// TODO: do we always book a block even if its transaction is not solid?
```

Comment:
Very good point. Although your own conflict is always known, even if the transaction is not solid. What you do not know are the parents of such conflict to properly create the conflict with the correct relationships in the ConflictDAG. It doesn't change the labels that this block carries though!
This should conern the component that it is actually creating the conflict in the ConflictDAG, as, by doing so, it has to also specify its parents.